File: support_tools/swigwin-3.0.12/Examples/test-suite/lua/smart_pointer_multi_runme.lua
require("import")	-- the import fn
import("smart_pointer_multi")	-- import lib into global
spm=smart_pointer_multi --alias

-- catching undefined variables
local env = _ENV -- Lua 5.2
if not env then env = getfenv () end -- Lua 5.1
setmetatable(env, {__index=function (t,i) error("undefined global variable `"..i.."'",2) end})

foo = spm.Foo()
foo.x = 5
assert(foo:getx() == 5)

bar = spm.Bar(foo)
spam = spm.Spam(bar)
grok = spm.Grok(bar)

assert(bar:getx() == 5)
assert(spam:getx() == 5)
spam.x = 7
assert(grok:getx() == 7)
grok.x = 10
assert(foo:getx() == 10)

File: support_tools/swigwin-3.0.12/Examples/test-suite/octave/smart_pointer_multi_runme.m
smart_pointer_multi

f = Foo();
b = Bar(f);
s = Spam(b);
g = Grok(b);

s.x = 3;
if (s.getx() != 3)
    error
endif

g.x = 4;
if (g.getx() != 4)
    error
endif

